pkg/sql/sem/tree/function_definition.go

```diff
@@ -67,17 +67,15 @@ func (fd *FunctionDefinition) String() string { return AsString(fd) }
 func (n *UnresolvedName) ResolveFunction(
 	searchPath sessiondata.SearchPath,
 ) (*FunctionDefinition, error) {
-	fn, err := n.normalizeFunctionName()
-	if err != nil {
-		return nil, err
+	if n.NumParts > 2 || len(n.Parts[0]) == 0 {
+		// n.NumParts > 2 would be catalog.schema.funname().
+		// We don't support those. (yet?)
+		return nil, pgerror.NewErrorf(pgerror.CodeSyntaxError,
+			"invalid function name: %q", ErrString(n))
```

Comment:
We will need this soon I think. Another thing we need while I'm thinking about it is namespaced types like `::pg_catalog.regclass`.